common/lib/capa/capa/util.py

```diff
@@ -65,24 +65,22 @@ def compare_with_tolerance(student_complex, instructor_complex, tolerance=defaul
         # `tolerance` both equal to infinity. Then, below we would have
         # `inf <= inf` which is a fail. Instead, compare directly.
         return student_complex == instructor_complex
+
+    if isinstance(instructor_complex, complex):
+        if instructor_complex.imag == 0:
+            instructor_complex = instructor_complex.real
+
+    if isinstance(student_complex, float) and isinstance(instructor_complex, float):
```

Comment:
student_complex can be only float for tests [here](https://github.com/edx/edx-platform/blob/dbc465a51871bd685dd925c23bf73b981e07abe6/common/lib/capa/capa/tests/test_util.py#L18)